interface.go

```diff
@@ -9,6 +9,9 @@ import (
 // HeaderFetcher aliases a function that is used to fetch an ExtendedHeader from store by height.
 type HeaderFetcher[H header.Header[H]] func(context.Context, uint64) (H, error)
 
+// HeadFetcher aliases a function that is used to get current network head.
+type HeadFetcher[H header.Header[H]] func(ctx context.Context) (H, error)
```

Comment:
I don't know why we call them Fetchers rather Getters. Let's call this one Getter and make an issue to rename another one.

Reply:
https://github.com/celestiaorg/go-fraud/issues/38